python/ray/data/dataset_pipeline.py

```diff
@@ -700,6 +682,325 @@ def __iter__(self):
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.map`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.map(fn, compute=compute, **ray_remote_args)
+        )
+
+    def map_batches(
+        self,
+        fn: BatchUDF,
+        *,
+        batch_size: Optional[int] = 4096,
+        compute: Union[str, ComputeStrategy] = None,
+        batch_format: str = "native",
+        fn_args: Optional[Iterable[Any]] = None,
+        fn_kwargs: Optional[Dict[str, Any]] = None,
+        fn_constructor_args: Optional[Iterable[Any]] = None,
+        fn_constructor_kwargs: Optional[Dict[str, Any]] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.map_batches`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.map_batches(
+                fn,
+                batch_size=batch_size,
+                compute=compute,
+                batch_format=batch_format,
+                fn_args=fn_args,
+                fn_kwargs=fn_kwargs,
+                fn_constructor_args=fn_constructor_args,
+                fn_constructor_kwargs=fn_constructor_kwargs,
+                **ray_remote_args,
+            )
+        )
+
+    def flat_map(
+        self,
+        fn: RowUDF,
+        *,
+        compute: Union[str, ComputeStrategy] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.flat_map`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.flat_map(fn, compute=compute, **ray_remote_args)
+        )
+
+    def filter(
+        self,
+        fn: RowUDF,
+        *,
+        compute: Union[str, ComputeStrategy] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[T]":
+        """Apply ``Dataset.filter`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.filter(fn, compute=compute, **ray_remote_args)
+        )
+
+    def add_column(
+        self,
+        col: str,
+        fn: Callable[["pandas.DataFrame"], "pandas.Series"],
+        *,
+        compute: Optional[str] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.add_column`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.add_column(col, fn, compute=compute, **ray_remote_args)
+        )
+
+    def drop_columns(
+        self,
+        cols: List[str],
+        *,
+        compute: Optional[str] = None,
+        **ray_remote_args,
+    ) -> "DatasetPipeline[U]":
+        """Apply ``Dataset.drop_columns`` to each dataset/window in this pipeline."""
+        return self.foreach_window(
+            lambda ds: ds.drop_columns(cols, compute=compute, **ray_remote_args)
+        )
+
+    def repartition(
```

Comment:
Are those still needed?

Reply:
I am not sure our deprecation policy (i.e. when should we delete code for deprecated APIs). But if they are not widely used by Ray users, I think we should just delete them.

Reply:
We're pre-GA and these have been deprecated for a long time (multiple releases), so I think that we should remove them before the 2.0 release.

Reply:
@jianoaix, @clarkzinzow - removed the deprecated APIs. Thanks.